model/src/main/proto/survey.proto

```diff
@@ -61,43 +60,64 @@ enum QuestionName {
   //  Corresponds to the market fit question.
   MARKET_FIT = 2;
 
-  // Corresponds to  the NPS question.
+  // Corresponds to the NPS question.
   NPS = 3;
 
-  // Corresponds to  the open-ended question for a NPS promoter.
+  // Corresponds to the open-ended question for a NPS promoter.
   PROMOTER_FEEDBACK = 4;
 
-  // Corresponds to  the open-ended question for a NPS passive.
+  // Corresponds to the open-ended question for a NPS passive.
   PASSIVE_FEEDBACK = 5;
 
-  // Corresponds to  the open-ended question for a NPS detractor.
+  // Corresponds to the open-ended question for a NPS detractor.
   DETRACTOR_FEEDBACK = 6;
 }
 
 // Represents the multiple choice answer available for a question.
-message SurveyAnswerOption {
-  // The ID of this answer option. It is a string value that uniquely identifies the answer option
-  // within the context of the question it belongs to.
-  string question_answer_id = 1;
+message SurveyQuestionOption {
+  // The ID of this question option. It is a string that uniquely identifies the question option
+  // within the context of a survey question. It is only unique within the scope of a survey.
+  string question_option_id = 1;
 
   // The semantic name of the question this answer option is tied to.
   QuestionName question_name = 2;
 
-  // The actual text value of the answer.
-  string answer = 3;
+  // The actual text value of the answer option.
+  string label = 3;
+
+  // The type of answer this option represents.
+  oneof answer {
+    // The enum representation of the type of user answer option.
+    UserTypeAnswer user_type = 4;
+
+    // The enum representation of the market fit answer option.
+    MarketFitAnswer market_fit = 5;
+
+    // The integer value representing the nps score option.
+    int32 nps = 6;
+  }
 }
 
 // Represents an answer that was selected by a user for a survey question.
 message SurveyUserAnswer {
   // The ID of the survey to which this answer is tied. This is a string value that uniquely
-  // identifies the survey.
+  // identifies the survey globally.
   string survey_id = 1;
 
   // Semantic name of the question the response is tied to.
   QuestionName question_name = 2;
 
   // The value of the answer the user gave.
-  string user_answer = 3;
+  oneof answer {
+    // The enum representation of the type of user answer selected.
+    UserTypeAnswer user_type = 3;
+
+    // The enum representation of the market fit answer selected.
+    MarketFitAnswer market_fit = 4;
+
+    // The integer value representing the nps score selected by the user.
+    int32 nps = 5;
```

Comment:
Perhaps ``nps_score``, instead, per the comment? Ditto above.

Reply:
I've updated both fields.